src/App.jsx

```diff
@@ -7,19 +7,35 @@ import database from './database.json';
 import './App.css';
 
 function App() {
+  const [productsList, setProductsList] = useState(database);
+
+  const upvoteProduct = (id) => {
+    const productListClone = [...productsList];
+    const elementIndex = productListClone.findIndex((el) => el.id === id);
+
+    productListClone[elementIndex] = {
+      ...productListClone[elementIndex],
+      upvotes: productListClone[elementIndex].upvotes + 1,
+    };
+
+    setProductsList(productListClone);
+  };
+
   return (
     <>
       <Header />
       <main>
         <ul className="product-list">
-          {database.map((product) => (
+          {productsList.map((product) => (
             <Product
+              id={product.id}
```

Comment:
We need to know the product's ID from the component, as the `upvoteProduct` needs it, so we are specifying it here